packages/workspace/src/workspace/workspace.ts

```diff
@@ -1147,14 +1171,16 @@ export const loadWorkspace = async (
     ),
     getReferenceSourcesIndex: async (envName = currentEnv()) => (await getWorkspaceState())
       .states[envName].referenceSources,
-    getReferenceTargetsIndex: async (envName = currentEnv()) => (await getWorkspaceState())
-      .states[envName].referenceTargets,
     getElementOutgoingReferences: async (id, envName = currentEnv()) => {
-      if (!id.isBaseID()) {
-        throw new Error(`getElementOutgoingReferences only support base ids, received ${id.getFullName()}`)
-      }
-      return await (await getWorkspaceState()).states[envName]
-        .referenceTargets.get(id.getFullName()) ?? []
+      const baseId = id.createBaseID().parent.getFullName()
+      const referencesTree = await (await getWorkspaceState()).states[envName].referenceTargets.get(baseId)
+
+      const idPath = id.createBaseID().path.join(ElemID.NAMESPACE_SEPARATOR)
+      return referencesTree === undefined ? []
+        : _.uniqBy(
+          Array.from(referencesTree.valuesWithPrefix(idPath)).flat(),
+          elemId => elemId.getFullName()
+        )
```

Comment:
if we want to avoid the inconsistency in tree_map, we can change this to
```suggestion
      if (referencesTree === undefined) {
        return []
      }
      const idPath = id.createBaseID().path.join(ElemID.NAMESPACE_SEPARATOR)
      const references = idPath === ''
        ? referencesTree.values()
        : referencesTree.valuesWithPrefix(idPath)
      return _.uniqBy(Array.from(references).flat(), elemId => elemId.getFullName())
```